rfd/0203-database-healthchecks.md

```diff
@@ -0,0 +1,697 @@
+      labels:
+        teleport.internal/resource-type: preset
+    spec:
+      enabled: true
+      match:
+        db_labels:
+          - name: '*'
+            values:
+            - '*'
+      interval: 10s
+      timeout: 5s
+      healthy_threshold: 2
+      unhealthy_threshold: 1
+    
+Out of an abundance of caution around this new feature, Alice wants to opt-out of health checks for prod databases.
+
+Alice uses `tctl edit health_check_config/default` to update the default settings and exclude prod databases:
+
+    kind: health_check_config
+    version: v1
+    metadata:
+      name: "default"
+      labels:
+        teleport.internal/resource-type: preset
+    spec:
+      enabled: true
+      match:
+        db_labels:
+          - name: '*'
+            values:
+            - '*'
+        db_labels_expression: "labels.env != `prod`"
+      interval: 10s
+      timeout: 5s
+      healthy_threshold: 2
+      unhealthy_threshold: 1
+
+#### User story: AWS RDS database enrollment
+
+A Teleport cluster admin is new to Teleport and would like to enroll an AWS RDS database with Teleport.
+
+- The user logs into Teleport, sees the "Enroll New Resource" button and clicks that.
+- They click on the "RDS PostgreSQL" tile - a guided database enrollment wizard.
+- After selecting an AWS integration, region, and VPC, the user sees the "Enroll RDS Database" page.
+- The user selects a database to enroll, and clicks "Next". 
+- On the next page, the user is asked to configure an IAM role, choose AWS subnets and security groups, and finally deploy a Teleport database service in ECS using those settings.
+- The user selects a public subnet and a security group that allows all outbound traffic, then clicks "Deploy".
+
+The page tells the user that it is waiting for the new DB agent to start and join the Teleport cluster:
+
+    Teleport is currently deploying a Database Service.
+    It will take at least a minute for the Database Service to be created and joined to your cluster.
+
+After the new agent joins, the page shows a success message:
+
+    Successfully created and detected your new Database Service.
+
+The page displays a new message below that:
+
+    Teleport is testing network connectivity between the Database Service and your database.
+
+After a short time, the page displays a message telling the user that the agent cannot reach the database endpoint:
+
+    The Database Service does not have connectivity to the database.
+    1. Check that the security groups you selected allow outbound TCP traffic from the Teleport Database Service to the database address on port 5432
+    2. Check that the database security groups allow inbound traffic from the Teleport Database Service on port 5432
+
+    Troubleshooting tips: <link to teleport docs>
+
+- The user realizes that they need to select an additional security group that the database allows inbounds traffic from.
+- They select the additional security group and click "Redeploy".
```

Comment:
We don't support redeployment yet do we, we'd have to build it? Should we keep integration of this into Discover flows for phase 2?